### ISSUE TITLE
Bejeweled 2 [NPUG30038] Game crashes either at end of loading bar or when fast fowarding

### ISSUE THREAD

**Issue by Jack54guythecoder** [hrydgard/ppsspp#19708]
### Game or games this happens in

Bejeweled 2 [NPUG30038]

### What area of the game / PPSSPP

Just hit the icon to play and either let the game finish loading or fast foward, and when doing the latter, a BSOD style screen appears.
![BSOD SCREEN](https://github.com/user-attachments/assets/ce5f2151-9047-49f5-94e4-523827c0f21e)

### What should happen

It should just work when fast fowarding as well

### Platform

Windows

### Mobile device model or graphics card (GPU)

NVIDIA GeForce RTX 3050 Laptop GPU running on driver 546.80.0.0

### PPSSPP version affected

Bejeweled 2 [NPUG30038]

### Last working version

a couple of builds before

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://www.ppsspp.org/devbuilds) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [X] Include logs or screenshots of issue.

**Comment by anr2me** [comment]
Probably a duplicate of https://github.com/hrydgard/ppsspp/issues/15304

**Comment by hrydgard** [comment]
Hm, I thought the compat.ini hack took care of it, and I don't know what might have changed here.. Also, it's strange that fast-forward actually plays a role, it shouldn't. Overall weird..

**Comment by sum2012** [comment]
@Jack54guythecoder does you extract all files ,then run PPSSPPWindows64.exe ?

**Comment by hrydgard** [comment]
(and if you do, then don't - PSP ISOs are not meant to be extracted before running, just run the ISO directly)

**Comment by Jack54guythecoder** [comment]
> edit: @Jack54guythecoder does you extract ppsspp_win_v1.18.1-313-gd21de81826.zip to all files ,then run PPSSPPWindows64.exe ?

Why would you ask this?
ofc I have it extracted to a folder, I am still in 309, but will download 313 later.

@hrydgard  Also, to answer your question, I have my games in CHD format.

**Comment by Jack54guythecoder** [comment]
UPDATE: Its still broken on v1.18.1-317

**Comment by hrydgard** [comment]
ok, very strange since the workaround should still work. but thanks for reporting

**Comment by sum2012** [comment]
@Jack54guythecoder can you try the game in ISO format ?

**Comment by Jack54guythecoder** [comment]
Will do.
You think there will be a difference there?

**Comment by Jack54guythecoder** [comment]
Changing to ISO DOES NOT make a difference.
My settings are here: 
[ppsspp.zip](https://github.com/user-attachments/files/18068982/ppsspp.zip)

**Comment by sum2012** [comment]
@Jack54guythecoder I don't have idea. I don't have the region on hand.
Please use v1.18.1-339. Screenshoot the bluescreen

**Comment by Jack54guythecoder** [comment]
Screenshooted the bluescreen.

![image](https://github.com/user-attachments/assets/39350642-580d-48c0-b2fc-468f2de62ba5)

**Comment by Jack54guythecoder** [comment]
There seems to be an CRC different from the others here.
https://report.ppsspp.org/game/NPUG30038_1.00
CRC b7488b47

**Comment by Jack54guythecoder** [comment]
Issue Report for Bejeweled 2 on PPSSPP
A user reported a crash issue with the game Bejeweled 2 [NPUG30038] on the PPSSPP emulator, occurring either at the end of the loading bar or when fast-forwarding. The user detailed their system specifications, including an NVIDIA GeForce RTX 3050 Laptop GPU and the Vulkan graphics backend.

Multiple comments followed, with suggestions to test different settings and formats (ISO vs. CHD), and to check if the problem persists in the latest builds. The owner, hrydgard, expressed confusion over the issue since a workaround was previously established. The user confirmed the ongoing problem across versions and noted a potential CRC discrepancy. Troubleshooting and collaboration continued among users and developers to resolve the issue.

**Comment by sum2012** [comment]
@hrydgard why the compat not active ? I sure don't spell mistake in the GameId in compat.ini

**Comment by anr2me** [comment]
@Jack54guythecoder could you try the artifacts from https://github.com/ANR2MERefork/ppsspp/actions/runs/12265058234?pr=18 to see whether it helped?

**Comment by Jack54guythecoder** [comment]
> @Jack54guythecoder could you try the artifacts from https://github.com/ANR2MERefork/ppsspp/actions/runs/12265058234?pr=18 to see whether it helped?

How do i get the artifact?

**Comment by hrydgard** [comment]
Good question. The problem here is that the compatibility flag for this game has not been activated, as @sum2012 observes.

Not sure why - is your assets/compat.ini broken or missing? 

Do you have `IgnoreCompatSettings` set under `[General]` in ppsspp.ini ?

**Comment by Jack54guythecoder** [comment]
> Good question @sum2012 . The problem here is indeed that the compatibility flag for this game has not been activated, as @sum2012 observes.
> 
> Not sure why - is your assets/compat.ini broken or missing?
> 
> Do you have `IgnoreCompatSettings` set under `[General]` in ppsspp.ini ?

Are you talking to me or who?

**Comment by Jack54guythecoder** [comment]
Huh???

It was like this the WHOLE TIME

![image](https://github.com/user-attachments/assets/fea5ce67-71e2-4654-8706-8ea2971f2e2a)
.
This is WEIRD.

**Comment by anr2me** [comment]
> > @Jack54guythecoder could you try the artifacts from https://github.com/ANR2MERefork/ppsspp/actions/runs/12265058234?pr=18 to see whether it helped?
> 
> How do i get the artifact?

Here is the direct link to the Windows x64 build artifact: https://github.com/ANR2MERefork/ppsspp/actions/runs/12265058234/artifacts/2302522206

**Comment by hrydgard** [comment]
@Jack54guythecoder yes it's very weird. It's clear from your crash report that the compatibility setting is not enabled.

So the only thing I can think of is that assets/compat.ini under your PPSSPP install directory must be missing or corrupted. Otherwise I really have no idea..

**Comment by Jack54guythecoder** [comment]
[compat.zip](https://github.com/user-attachments/files/18131525/compat.zip)
The thing is, its there.

**Comment by Jack54guythecoder** [comment]
![image](https://github.com/user-attachments/assets/8ba74c58-c23f-4859-81af-d24c4598c304)

**Comment by anr2me** [comment]
May be we should shows active compat at this screen too, i'm not sure whether the compat.ini is active or not but the game works fine even when fast forwarded to this screen.
![image](https://github.com/user-attachments/assets/760975a9-f0f6-4cbc-b0b1-bd97550e1fed)

PS: This is from a newly extracted ppsspp devbuild, so it uses the default settings.

**Comment by Jack54guythecoder** [comment]
v1.18.1-405 works good.

So what can I do now?

**Comment by Jack54guythecoder** [comment]
Any ideas as to what can be done?

**Comment by hrydgard** [comment]
If 405 works good, I don't know that there's anything to do. I'm still baffled though.

**Comment by Jack54guythecoder** [comment]
> If 405 works good, I don't know that there's anything to do. I'm still baffled though.

The thing is, the new folder works good, but the previous one DOES NOT work.
Should I transfer something from the new folder into the old one?

**Comment by hrydgard** [comment]
Oh, then it really seems it's some setting. Could you post the PSP/SYSTEM/ppsspp.ini file from your non-working folder here? (Just drag it into the text to upload, might need to zip it up for the file to get accepted)

**Comment by Jack54guythecoder** [comment]
[ppsspp.zip](https://github.com/user-attachments/files/18138590/ppsspp.zip)

@hrydgard  Here it is, the non-working one.

**Comment by anr2me** [comment]
I've tried using your ppsspp.ini, but the game is still running fine without any issue when fast forwarded.
May be because of different CRC?

**Comment by Jack54guythecoder** [comment]
@anr2me if you got to the REPORT PPSSPP site, then you will see various people with the same CRC as mine.

**Comment by anr2me** [comment]
The replacing the ppsspp.ini on the new folder with the broken one, and see whether it also works good mine.
if it also works, may be something else is causing it, like an existing savedata may be...

**Comment by Jack54guythecoder** [comment]
> Try replacing the ppsspp.ini on the new folder with the broken one, and see whether the ppsspp on the new folder also works good like mine. if it also works, may be something else is causing it, like an existing savedata may be...

I replaced the .ini file from the new one with broken one, IT WORKED.

THEN HOW IN GODS NAME DOES IT NOT WORK ON THE OLD FOLDER?

I DO NOT think it can be a save data problem personally.

**Comment by anr2me** [comment]
Yeah, i tried using savedata from gamespot and the game still works fine when fast forwarded

**Comment by Jack54guythecoder** [comment]
> Well, i tried using savedata from gamespot and the game still works fine when fast forwarded. You can try copying your old memstick folder to overwrite the new one to see whether it was a file within the memstick folder causing it.

This did not work AT ALL.

**Comment by anr2me** [comment]
If using your old memstick folder on the new ppsspp folder didn't works, i still suspects it was due to your old savedata.
Can you upload your zipped savedata folder here?

**Comment by Jack54guythecoder** [comment]
[NPUG30038DATA00.zip](https://github.com/user-attachments/files/18173848/NPUG30038DATA00.zip)
Here it is.

**Comment by Jack54guythecoder** [comment]
WEIRD UPDATE: Changing psp cpu clock up to 200% fixes the crash for some reason.

**Comment by anr2me** [comment]
Changing the CPU clock is one of the workaround, as mentioned on the other similar issue at https://github.com/hrydgard/ppsspp/issues/15304#issuecomment-1013938950

**Comment by anr2me** [comment]
Okay. it seems your savedata really did cause the crash since it works fine with savedata from gamespot)
![image](https://github.com/user-attachments/assets/62bfdd5f-d8ea-4d40-a5e1-c764eb7f257e)

May be your savedata got corrupted or something, or may be it was saved using a PPSSPP version that have saving issue.
I would suggest using an older version of PPSSPP that doesn't crash and try to resave it, or using a different emulator like JPCSP to resave it.

**Comment by Jack54guythecoder** [comment]
Ye, tho in the end, I might delete the save altogether, havent progressed much, so it doesnt matter to me.

**Comment by Jack54guythecoder** [comment]
FINAL: I just put cpu clock to 200 and works fine, I think i can close this maself

**Comment by Jack54guythecoder** [comment]
UPDATE ON ISSUE> It seems like every time i make progress, save, then come back, regardless of updates to the emu, the game crashes like before.

**Comment by anr2me** [comment]
Changing the CPU clock isn't a proper fix, it's just a workaround for timing issue. 

It might works with 200 mhz, but sometimes it need a different value like 50 or 300.

**Comment by Jack54guythecoder** [comment]
Any update on this?

**Comment by sum2012** [comment]
@Jack54guythecoder I re-test, sometime error
